scripts/build-windows/kcl-fmt.go

```diff
@@ -6,54 +6,85 @@
 package main
 
 import (
+	"fmt"
 	"os"
+	"os/exec"
 	"path/filepath"
-	"syscall"
-	"unsafe"
 )
 
 func main() {
-	// kclvm -m kclvm ...
+	// python3 -m kclvm ...
+
 	var args []string
-	args = append(args, os.Args[0])
-	args = append(args, "-m", "kclvm.tools.format")
+	args = append(args, "/C", "python3", "-m", "kclvm.tools.format")
 	args = append(args, os.Args[1:]...)
+
 	os.Exit(Py_Main(args))
 }
 
-var (
-	python39_dll = syscall.NewLazyDLL(findKclvm_dllPath())
-	proc_Py_Main = python39_dll.NewProc("Py_Main")
-)
-
-// int Py_Main(int argc, wchar_t **argv)
 func Py_Main(args []string) int {
-	c_args := make([]*uint16, len(args)+1)
-	for i, s := range args {
-		c_args[i] = syscall.StringToUTF16Ptr(s)
+	inputPath, err := os.Executable()
+	if err != nil {
+		fmt.Println("Input path does not exist")
+		os.Exit(1)
+	}
+	kclvm_install_dir_bin := filepath.Dir(inputPath)
+	Install_Kclvm(kclvm_install_dir_bin)
+	kclvm_install_dir := filepath.Dir(kclvm_install_dir_bin)
+
+	cmd := exec.Command("cmd", args...)
+
+	Set_Env(kclvm_install_dir, cmd)
+
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+
+	err = cmd.Run()
+	if err != nil {
+		fmt.Println("exec failed:", err)
+		os.Exit(1)
 	}
-	ret, _, _ := proc_Py_Main.Call(uintptr(len(args)), uintptr(unsafe.Pointer(&c_args[0])))
-	return int(ret)
+	return 0
 }
 
-func findKclvm_dllPath() string {
-	kclvmName := "python39.dll"
+func Install_Kclvm(installed_path string) {
+	// Check if Python3 is installed
+	cmd := exec.Command("cmd", "/C", "where python3")
+	_, err := cmd.Output()
+	if err != nil {
+		fmt.Println("Python3 is not installed, details: ", err)
```

Comment:
Should the promotion information be output to stderr? All other places where `os.Exit(1)` appears should be considered this way.

Reply:
fixed